agent/health_endpoint_test.go

```diff
@@ -1418,13 +1418,22 @@ func TestHealthConnectServiceNodes(t *testing.T) {
 }
 
 func TestHealthIngressServiceNodes(t *testing.T) {
+	t.Run("no streaming", func(t *testing.T) {
+		testHealthIngressServiceNodes(t, ` rpc { enable_streaming = false } use_streaming_backend = false `)
+	})
+	t.Run("cache with streaming", func(t *testing.T) {
+		testHealthIngressServiceNodes(t, ` rpc { enable_streaming = true } use_streaming_backend = true `)
+	})
+}
+
+func testHealthIngressServiceNodes(t *testing.T, agentHCL string) {
+	t.Helper()
```

Comment:
```suggestion
```

This function contains all the logic for the test, so I think this is a "test case" function not a test helper function. Using `t.Helper` in a test case does not help us identify the case, because we already know which case is which from the name. It can actually make the failures harder to debug since we won't get the line number for the failed assertion, and it's not always obvious which line failed.